src/MemPool.jl

```diff
@@ -100,6 +100,14 @@ function approx_size(xs::AbstractArray{String})
     s + 4 * length(xs)
 end
 
+function approx_size(s::String) 
+    sizeof(s)+8
+end
+
+function approx_size(s::Symbol) 
+    0
```

Comment:
This should be at least 8 bytes, from testing with `serialize`. I don't want us to have to do `string(s)` to figure out the length, though, so let's just do 8.

Reply:
The size has to correlate with the length, no?

I just added the exact same implementation as for String and it is fast. Wouldn't that be better than just 8?

Here is a quick and dirty experiment:

```julia
# Somewhat irregular difference for small sizes
julia> map(1:10) do x
       s = String(rand('a':'z', x))
       pbstr = PipeBuffer()
       pbsym = PipeBuffer()
       serialize(pbstr, s)
       serialize(pbsym, Symbol(s))
       Base.summarysize(pbstr) - Base.summarysize(pbsym)
       end
10-element Vector{Int64}:
 2
 0
 0
 0
 0
 0
 0
 1
 1
 1

# Large sizes seem pretty consistent
julia> map(10000:10005) do x
       s = String(rand('a':'z', x))
       pbstr = PipeBuffer()
       pbsym = PipeBuffer()
       serialize(pbstr, s)
       serialize(pbsym, Symbol(s))
       Base.summarysize(pbstr) - Base.summarysize(pbsym)
       end
6-element Vector{Int64}:
 5
 5
 5
 5
 5
 5

julia> map(1:100:100000) do x
       s = String(rand('a':'z', x))
       pbstr = PipeBuffer()
       pbsym = PipeBuffer()
       serialize(pbstr, s)
       serialize(pbsym, Symbol(s))
       Base.summarysize(pbstr) - Base.summarysize(pbsym)
       end |> extrema
(1, 5)
``´

Reply:
Oh yeah, I just meant something other than returning `0` (because `Symbol` has size of the wire). This looks good!